app/controllers/api/sessions_controller.rb

```diff
@@ -0,0 +1,61 @@
+module Api
+  class SessionsController < ApplicationController
+    prepend_before_action :skip_session_load
+    prepend_before_action :skip_session_expiration
+    prepend_before_action :skip_devise_hooks
+
+    respond_to :json
+
+    def show
+      render json: { active: active?, timeout: timeout }
+    end
+
+    def update
+      analytics.session_kept_alive if active?
+      update_last_request_at
+      render json: { active: active?, timeout: timeout }
+    end
+
+    def destroy
+      analytics.session_timed_out
+      sign_out
+      render json: { redirect: root_url(request_id: sp_session[:request_id], timeout: :session) }
+    end
+
+    private
+
+    def skip_devise_hooks
+      request.env['devise.skip_timeout'] = true
+      request.env['devise.skip_trackable'] = true
+    end
+
+    def disable_caching
+      super
+      response.headers['Etag'] = ''
+    end
+
+    def active?
+      timeout.future?
+    end
+
+    def timeout
+      if last_request_at.present?
+        Time.zone.at(last_request_at + User.timeout_in).utc
+      else
+        Time.zone.now.utc
```

Comment:
`Time.zone` is UTC already (in most cases outside of zonebie specs)... so the `.utc` on there redundant. Another way to do this in one pass:
```suggestion
        ActiveSupport::TimeZone['UTC'].at(last_request_at + User.timeout_in)
      else
        ActiveSupport::TimeZone['UTC'].now
```

Reply:
I wonder if it's necessary to bother with timezones at all here, since assuming Rails would render the JSON as ISO8601 format, the parsing on the client-side would be able to convert it accurately all the same.

Reply:
Updated to drop UTC in ff3f754.